ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/attestation/AttestationManager.java

```diff
@@ -112,7 +113,10 @@ private void onBlockImported(final ImportedBlockEvent blockImportedEvent) {
                               "Failed to process pending attestation dependent on " + blockRoot,
                               err));
             });
-    block.getMessage().getBody().getAttestations().forEach(aggregatingAttestationPool::remove);
+  }
+
+  public SafeFuture<AttestationProcessingResult> onAttestation(Attestation attestation) {
+    return onAttestation(ValidateableAttestation.fromAttestation(attestation));
   }
```

Comment:
I probably wouldn't add this method.  We want to create the `ValidatableAttestation` only at the key entry points but this makes it much simpler to get that wrong and create a new `ValidatableAttestation` without thinking about it and then we'll lose the context it tracks.